### PR TITLE
Task/Build Featured Tours Section on Home Page

### PR DESCRIPTION
**Pull request by justjohnd**
Purpose:
Home page to have a "Featured Tours" section, that will be populated by any three "Tours" posts the client desires. Client must be able to easily change the tour posts. See https://www.muchbetteradventures.com/ "Popular Private Trips" section for reference.

Pull request to to produce functionality, not the final styled design of the sections, nor any working hyperlinks.

Note that code has been added to register a "featured-tours" sidebar (widget), though this prospect was temporarily shelved in lieu of the solution below. 

Method:
-  Add theme support to functions.php to be able to get thumbnail image data from database
- Query all posts for Category: Tours
- Add parameter to display only three posts
- Add parameter to display only posts flagged with a Sticky
- Use while loop to display query posts results
- For testing purposes, display the thumbnail image and post title
- Place each post result inside a div with a bootstrap grid class to make responsive
- Restore $post global to current position in main query (https://developer.wordpress.org/reference/functions/wp_reset_postdata/)

- Place all of the above in a function in functions.php
- Call the function on home page
- Add minor styling

Other Misc.:

- Hide the_title, and make it an h2
- Remove the edit button on the page (only visible when logged in as admin, but I don't use it)

Notes: 
- .featured-tours ul {} styles entirely unnecessary when using bootstrap
- If possible, it may be better to control image dimensions in css versus using wp methods inside functions.php
- Is functions.php the best place to keep this code?

Next Steps:
- Clone the "Recent Posts" widgets and modify with the above code. Then client can easily change the widget location if they want to modify the design of the page
- Change the Featured Tours section to a carousel to be able to add more posts

### CODE REVIEW

File: wp-content/themes/ski-tours/front-page.php [hunk 1]
```diff
@@ -15,24 +15,38 @@
...
-		endwhile; // End of the loop.
-		?>
-		<p>Template: front-page</p>
-	</main><!-- #main -->
+</main><!-- #main -->
 
 <?php
 get_sidebar();
-get_footer();
+get_footer();
```
Comment (eliotc1986) [inline wp-content/themes/ski-tours/front-page.php:52]: @justjohnd Make sure to turn on add new line to file in vs code

File: wp-content/themes/ski-tours/front-page.php [hunk 2]
```diff
@@ -15,24 +15,38 @@
...
+      <div class="row">
+        <?php echo ski_tours_recent_posts(); ?>
+      </div>  
+    </div>
+
+  <!-- Currently inactive widget for featured tours. Using the above function to enable thumbnails -->
+  <!-- <?php if ( is_active_sidebar( 'featured-tours' ) ) : ?>
+  <div div class="featured-tours container px-5">
+    <?php dynamic_sidebar( 'featured-tours' ); ?>
+  </div> -->
```
Comment (eliotc1986) [inline wp-content/themes/ski-tours/front-page.php:40]: @justjohnd nit: can this be removed?

File: wp-content/themes/ski-tours/functions.php [hunk 3]
```diff
@@ -187,3 +201,22 @@ function ski_tours_scripts() {
 	require get_template_directory() . '/inc/jetpack.php';
 }
 
+// Display tour posts tagged "sticky"
+function ski_tours_recent_posts() {
+		$stickies = get_option( 'sticky_posts' );
```
Comment (eliotc1986) [inline wp-content/themes/ski-tours/functions.php:206]: @justjohnd maybe you can change this to `get_ski_...`
I reccommend using `$sticky_posts` as is.

File: wp-content/themes/ski-tours/style.css [hunk 4]
```diff
@@ -1176,6 +1177,19 @@ textarea {
...
+.page-id-1064 .entry-title {
 	display: none;
+}
+
+.featured-tours ul {
+	text-align: left;
+	margin: 0;
+	padding: 0;	
+	justify-content: space-between;
+	list-style-type: none;
```
Comment (eliotc1986) [inline wp-content/themes/ski-tours/style.css:1189]: @justjohnd you can use bootstraps utility class for unstlyed list `list-unstyled` `.text-left`?

Also I think if you don't use display: flex, other flex properties won't register,

`flex: 1`

**Review by eliotc1986 — COMMENTED** [review]
Good work!

Since this task is still WIP, I recommend using the feat/child branching system.

File: wp-content/themes/ski-tours/front-page.php (hunk 1 above)
Comment (justjohnd) [reply wp-content/themes/ski-tours/front-page.php:46]: Fix 586ea92fbfc4a1cddd179922446736a896b79350s

File: wp-content/themes/ski-tours/functions.php [hunk 5]
```diff
@@ -187,3 +201,22 @@ function ski_tours_scripts() {
...
+					<?php the_post_thumbnail('ski-tours-recent-thumbnails'); ?>
+				</a>
+				<h4>
+					<a href="<?php esc_url(the_permalink()); ?>">
+						<?php esc_html(the_title()); ?>
+					</a>
+				</h4>
+			</div>
+<?php endwhile;
+    wp_reset_postdata();
```
Comment (justjohnd) [inline wp-content/themes/ski-tours/functions.php:221]: Fix https://github.com/justjohnd/organize-backcountry/pull/4/commits/4e2db56d4eed84ab9f45f3986d74bcf07f47cc32

File: wp-content/themes/ski-tours/style.css [hunk 6]
```diff
@@ -1176,6 +1177,11 @@ textarea {
...
+.page-id-1064 .entry-title {
 	display: none;
-}
\ No newline at end of file
+}
+
+.size-ski-tours-recent-thumbnails {
+	height: 100%;
+	object-fit: cover;
+}
```
Comment (justjohnd) [inline wp-content/themes/ski-tours/style.css:1187]: Fix https://github.com/justjohnd/organize-backcountry/pull/4/commits/a5a70d7c0501f517e0e5df47edfbf26130655b31

File: wp-content/themes/ski-tours/front-page.php (hunk 2 above)
Comment (eliotc1986) [reply wp-content/themes/ski-tours/front-page.php]: Fixed here: 586ea92

File: wp-content/themes/ski-tours/style.css (hunk 4 above)
Comment (eliotc1986) [reply wp-content/themes/ski-tours/style.css]: Fixed here: a5a70d7c0501f517e0e5df47edfbf26130655b31

File: wp-content/themes/ski-tours/functions.php (hunk 3 above)
Comment (eliotc1986) [reply wp-content/themes/ski-tours/functions.php]: Fixed here: 4e2db56d4eed84ab9f45f3986d74bcf07f47cc32

File: wp-content/themes/ski-tours/front-page.php (hunk 1 above)
Comment (eliotc1986) [reply wp-content/themes/ski-tours/front-page.php:45]: Fixed here: 2ded2ecaeaa342cf73007e2d7d37bb73de305581

Comment (eliotc1986) [reply wp-content/themes/ski-tours/front-page.php:45]: @justjohnd This works. BTW, this setting will only apply for this project workspace, so I recommend you do this setting under `User` instead and revert this change.

File: wp-content/themes/ski-tours/functions.php (hunk 3 above)
Comment (eliotc1986) [reply wp-content/themes/ski-tours/functions.php]: @justjohnd nice, looks better. [Nit] I still recommend changing `ski_tours_featured_posts` to `get_ski_tours_featured_posts`

**Pull request merged**